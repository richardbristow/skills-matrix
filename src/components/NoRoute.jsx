import React from 'react';
import { Link } from 'react-router-dom';
// eslint-disable-next-line no-unused-vars
import styled from 'styled-components/macro';

import StyledMain from '../shared/StyledMain';

const StyledNoRoute = styled(StyledMain)`
  text-align: center;
`;

const NoRoute = () => {
  return (
    <StyledNoRoute>
      <h1>404</h1>
      <p>Page does not exist.</p>
      <p>
        Do you want to go back&nbsp;
        <Link to="/">home</Link>
      </p>
    </StyledNoRoute>
  );
};

export default NoRoute;
