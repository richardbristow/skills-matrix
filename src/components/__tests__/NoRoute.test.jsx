import React from 'react';
import { shallow } from 'enzyme';
import NoRoute from '../NoRoute';

describe('NoRoute', () => {
  it('should render the NoRoute component without errors', () => {
    const wrapper = shallow(<NoRoute />);
    expect(wrapper.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
});
