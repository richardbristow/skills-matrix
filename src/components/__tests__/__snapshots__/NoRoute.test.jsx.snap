// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoRoute should render the NoRoute component without errors 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NoRoute />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h1>
          404
        </h1>,
        <p>
          Page does not exist.
        </p>,
        <p>
          Do you want to go back 
          <Link
            to="/"
          >
            home
          </Link>
        </p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "404",
        },
        "ref": null,
        "rendered": "404",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Page does not exist.",
        },
        "ref": null,
        "rendered": "Page does not exist.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Do you want to go back ",
            <Link
              to="/"
            >
              home
            </Link>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Do you want to go back ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "home",
              "to": "/",
            },
            "ref": null,
            "rendered": "home",
            "type": [Function],
          },
        ],
        "type": "p",
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "text-align: center; padding-top: 60px;",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h1>
            404
          </h1>,
          <p>
            Page does not exist.
          </p>,
          <p>
            Do you want to go back 
            <Link
              to="/"
            >
              home
            </Link>
          </p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "404",
          },
          "ref": null,
          "rendered": "404",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Page does not exist.",
          },
          "ref": null,
          "rendered": "Page does not exist.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Do you want to go back ",
              <Link
                to="/"
              >
                home
              </Link>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Do you want to go back ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "home",
                "to": "/",
              },
              "ref": null,
              "rendered": "home",
              "type": [Function],
            },
          ],
          "type": "p",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "text-align: center; padding-top: 60px;",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
